Treat errors from Frgaal compiler as errors, not as warnings

Errors reported from Frgaal compiler are, parsed by `sbt` as warnings rather than errors. For example:
```
[info] compiling 1 Java source to /home/pavel/dev/enso/engine/runtime/target/scala-2.13/test-classes ...
[warn] Unexpected javac output: warning: The following options were not recognized by any processor: '[truffle.dsl.SuppressWarnings]'
[warn] /home/pavel/dev/enso/engine/runtime/src/test/java/org/enso/interpreter/test/caches/ModuleCacheTest.java:50: error: serialize(EnsoContext,CachedModule) has protected access in ModuleCache
[warn]     byte[] arr = module.getCache().serialize(ensoCtx, cm);
[warn]                                   ^
[warn] /home/pavel/dev/enso/engine/runtime/src/test/java/org/enso/interpreter/test/caches/ModuleCacheTest.java:53: error: deserialize(EnsoContext,byte[],Metadata,TruffleLogger) has protected access in ModuleCache
[warn]     var cachedIr = module.getCache().deserialize(ensoCtx, arr, meta, null);
[warn]                                     ^
[warn] /home/pavel/dev/enso/engine/runtime/src/test/java/org/enso/interpreter/test/caches/ModuleCacheTest.java:76: error: serialize(EnsoContext,CachedModule) has protected access in ModuleCache
[warn]     byte[] arr = module.getCache().serialize(ensoCtx, cm);
[warn]                                   ^
[warn] /home/pavel/dev/enso/engine/runtime/src/test/java/org/enso/interpreter/test/caches/ModuleCacheTest.java:79: error: deserialize(EnsoContext,byte[],Metadata,TruffleLogger) has protected access in ModuleCache
[warn]     var cachedIr = module.getCache().deserialize(ensoCtx, arr, meta, null);
[warn]                                     ^
[warn] 4 errors
[warn] 1 warning.
[warn] frgaal exited with exit code 1
[error] (Test / compileIncremental) javac returned non-zero exit code
[error] Total time: 3 s, completed Dec 11, 2023, 7:00:16 PM

```
This can lead to some confusions.

Let's modify javac output processor in `FrgaalCompiler.scala` such that errors are correctly recognized.

--- a/project/FrgaalJavaCompiler.scala
+++ b/project/FrgaalJavaCompiler.scala
@@ -22,6 +22,8 @@ import java.nio.file.{Path, Paths}
 import scala.sys.process.Process
 import scala.util.Using
 import java.io.FileWriter
+import xsbti.Severity
+import xsbti.Position
 
 object FrgaalJavaCompiler {
   private val ENSO_SOURCES = ".enso-sources"
@@ -254,6 +256,29 @@ object FrgaalJavaCompiler {
         exitCode = Process(exe +: forkArgs, cwd) ! javacLogger
       } finally {
         javacLogger.flush("frgaal", exitCode)
+      }
+      if (exitCode != 0) {
+        class FrgaalPosition extends xsbti.Position {
+          def line(): java.util.Optional[Integer]   = java.util.Optional.empty()
+          def lineContent()                         = "Frgaal errors"
+          def offset(): java.util.Optional[Integer] = java.util.Optional.empty()
+          def pointer(): java.util.Optional[Integer] =
+            java.util.Optional.empty()
+          def pointerSpace(): java.util.Optional[String] =
+            java.util.Optional.empty()
+          def sourceFile(): java.util.Optional[java.io.File] =
+            java.util.Optional.empty()
+          def sourcePath(): java.util.Optional[String] =
+            java.util.Optional.empty()
+        }
+
+        class FrgaalProblem extends xsbti.Problem {
+          def category()           = "Compiler error"
+          def severity(): Severity = Severity.Error
+          def message()            = "Error in frgaal compilation"
+          def position(): Position = new FrgaalPosition()
+        }
+        reporter.log(new FrgaalProblem())
       }
       // We return true or false, depending on success.
       exitCode == 0

